pkg/network/vlan/vlan.go

```diff
@@ -194,89 +134,53 @@ func (v *Vlan) AddLocalArea(id int, cidr string) error {
 	if err := iface.EnsureRouteViaGateway(cidr); err != nil {
 		return fmt.Errorf("ensure %s to route via gateway failed, error: %w", cidr, err)
 	}
-	// update routes of bridge
-	if v.NIC().Index() == v.mgmtNetwork.NIC().Index() {
-		return v.bridge.Fetch()
-	}
 
 	return nil
 }
 
-func (v *Vlan) RemoveLocalArea(id int, cidr string) error {
-	if v.nic == nil {
-		return fmt.Errorf("physical nic vlan network")
+func (v *Vlan) RemoveLocalArea(vid uint16, cidr string) error {
+	if v.uplink == nil {
+		return fmt.Errorf("bridge %s hasn't attached an uplink", v.bridge.Name)
 	}
 
-	if err := v.nic.DelBridgeVlan(uint16(id)); err != nil {
-		return fmt.Errorf("remove bridge vlan %d failed, error: %w", id, err)
+	ok, index := v.findVid(vid)
+	if !ok {
+		return nil
 	}
 
+	if err := v.uplink.DelBridgeVlan(vid); err != nil {
+		return fmt.Errorf("remove bridge vlanconfig %d failed, error: %w", vid, err)
+	}
+	v.vids = append(v.vids[:index], v.vids[index+1:]...)
```

Comment:
It's not out of range and a normal way to delete item from slice.

Reply:
I'm not sure of the reason not to allow all VLAN directly.

Reply:
@tjjh89017 It sounds like a good idea.

Reply:
Because the netlink package could add only one VLAN ID at one time, allow all VLAN directly means that the program will execute thousands of command when creating or updating a bond. It sounds not elegant.

Reply:
Because the netlink package could add only one VLAN ID at one time, allow all VLAN directly means that the program will execute thousands of commands when creating or updating a bond. It sounds not elegant.